nixos/modules/services/databases/pgbouncer.nix

```diff
@@ -3,14 +3,20 @@ let
   cfg = config.services.pgbouncer;
 
   settingsFormat = pkgs.formats.ini { };
-  configFile = settingsFormat.generate "pgbouncer.ini" cfg.settings;
+  configFile = settingsFormat.generate "pgbouncer.ini"
+    (lib.filterAttrsRecursive (_: v: v != null) cfg.settings);
   configPath = "pgbouncer/pgbouncer.ini";
 in
 {
   imports = [
-    (lib.mkRenamedOptionModule
+    (lib.mkRemovedOptionModule
       [ "services" "pgbouncer" "logFile" ]
-      [ "services" "pgbouncer" "settings" "pgbouncer" "log_file" ])
+      ''
+        `services.pgbouncer.logFile` has been removed, use `services.pgbouncer.settings.pgbouncer.logfile`
+        instead.
+        Please note that the new option expects an absolute path
+        whereas the old option accepted paths relative to pgbouncer's home dir.
+      '')
```

Comment:
ah shit this got buried in my notifications.

The answer is a clear yes though: the `mkRenamedOptionModule` is wrong here and you'll end up with a broken config if you set it for yourself since it expected a relative path whereas the `settings.pgbouncer.log_file` option expects an absolute path.